sdk/identity/identity/src/util/tracingUtils.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { TracerProxy, Span, SupportedPlugins, GetTokenOptions } from "@azure/core-http";
+
+/**
+ * Creates a span with the given name and spanOptions.
+ *
+ * @param name Name of the span
+ * @param spanOptions Span Options
+ */
+export function createSpan(name: string, spanOptions: any): Span {
+  const tracer = TracerProxy.getTracer();
+  const span = tracer.startSpan(name, spanOptions);
+  if (tracer.pluginType !== SupportedPlugins.NOOP && (spanOptions && spanOptions.parent)) {
+    spanOptions = { ...spanOptions, parent: span };
```

Comment:
This will only update the local variable `spanOptions` and not the actual `spanOptions` on the outer options of type `GetTokenOptions` which is what we want right?

Updating the local variable serves no purpose.

I would suggest the below
- Update `createSpan` to take options of type `GetTokenOptions` as the second parameter
- Then the above line would become `options.spanOptions = { ...options.spanOptions, parent: span };

Reply:
On second thoughts, I realize that updating the `parent` on the spanOptions might not be a good idea to do inside `createSpan`.

There are cases in this PR where we pass the same options (of type `GetTokenOptions`) to multiple methods which internally create more spans. Having the options updated inside `createSpan` will mess these parallel calls that are happening.

Essentially, we might have to clone the options each time it gets passed around.